modules/scrap_towny_ffa/biters.lua

```diff
@@ -112,28 +112,28 @@ function Public.unit_groups_start_moving()
 end
 
 function Public.swarm(town_center, radius)
-	local radius = radius or 32
-	local town_center = town_center or roll_market()
-	if not town_center or radius > 510 then return end
+	local r = radius or 32
+	local tc = town_center or roll_market()
+	if not tc or radius > 510 then return end
 
 	-- skip if we have to many swarms already
 	local count = table_size(global.towny.swarms)
 	local towns = #global.towny.town_centers
 	if count > 3 * towns then return end
 
-	local market = town_center.market
+	local market = tc.market
 	local surface = market.surface
 
 	-- find a spawner
-	local spawner = get_random_close_spawner(surface, market, radius)
+	local spawner = get_random_close_spawner(surface, market, r)
 	if not spawner then
-		radius = radius + 16
+		r = r + 16
 		local future = game.tick + 1
 		-- schedule to run this method again with a higher radius on next tick
 		if not tick_schedule[future] then tick_schedule[future] = {} end
 		tick_schedule[future][#tick_schedule[future] + 1] = {
 			callback = 'swarm',
```

Comment:
This will still not work. When on_tick calls the tick_schedule and tries to run the function `swarm` it will return nil.
```suggestion
			callback = 'Public.swarm',
```

Reply:
Looks like it is resolved.